### DISCUSSION TITLE
Inference on cropped images

### DISCUSSION THREAD

**Discussion by xavan83** [ultralytics/ultralytics#21902] [Q&A]
I use my custom dataset to train a model (yolo, rt-detr,...) using coco pretrained weight (I don't have enough images to train from scratch).
I use augmentation (rotation, zoom in, zoom out, multi-scale size training,etc).
I am getting pretty good result.

I have a lot of misclassification with Frigate regarding my lawnmower (class not in coco one).
I collect false positive and false negative snapshots from Frigate to try to enhance my model performance.
When I run inference locally using my test data and can not (never) reproduce that problem
The only way to reproduce it is to take a 320*320 crop of a labelled object and to run inference on it. 
I try to run a 640*640 model size but it is too much for my igpu.

This behavior occurs with all the models (and each model size).

Is it an expected behavior? Do I need to integrate cropped images to my dataset? Does a specific augmentation can address that?

Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xavan83, thanks for starting the discussion and for using Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In your case, it would be helpful to include:
- Exact `ultralytics` version and model weights used (e.g., YOLO11/RT-DETR variant and size)
- The precise inference command or Python snippet used locally vs. in Frigate
- A small set of representative images (original + 320x320 crops) and expected vs. actual predictions
- Any preprocessing/resizing settings in Frigate that differ from your local tests
- Training command, key hyperparameters, and sample training logs

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing a few misclassified examples and how they differ from your training distribution would also help us guide you 🎯.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Yes—this is expected: a 320×320 object crop changes scale and removes context vs. your full‑frame training distribution, so misclassifications can appear. To align train/deploy, fine‑tune a few epochs on object‑level chips at the same resolution you use in Frigate (e.g., 320) and include some background‑only negatives; you can quickly generate chips with the ObjectCropper workflow shown in [Object Cropping using Ultralytics YOLO11](https://docs.ultralytics.com/guides/object-cropping/):

```bash
yolo solutions crop source=path/to/images model=yolo11n.pt classes="[your_class_id]" crop_dir=crops
```

Also ensure your deployment resize matches training (set `imgsz=320`, consider `rect=True` to preserve aspect ratios) as noted in [Insights on Model Evaluation and Fine‑Tuning](https://docs.ultralytics.com/guides/model-evaluation-insights/). If Frigate is center‑cropping, note that `CenterCrop` is classification‑only and not part of the detection pipeline; see the [CenterCrop reference](https://docs.ultralytics.com/reference/data/augment/#ultralytics.data.augment.CenterCrop).

**Reply by xavan83** [reply]
This means my custom dataset needs only background images + cropped images?
Or is it still useful to have complete image scene like a full parking lot or garden?
Is it specific to yolo11 behavior?

**Comment by Y-T-G** [comment]
You can fine-tune YOLOE if your dataset is small. It would probably work better:

https://docs.ultralytics.com/models/yoloe/#__tabbed_1_1

**Reply by xavan83** [reply]
YOLOE seems to be used for segmentation, my use case is object detection

**Reply by Y-T-G** [reply]
It can be used for both. The link I provided shows how to fine-tune for both object detection and instance segmentation.
